### PR TITLE
features for stock command

### PR DESCRIPTION
**Pull request by uberscientist**
The supreme features laid out before your unworthy eyes, cast your :+1:s into the goblet of fire

### CODE REVIEW

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/21/

**Build Log**
_last 25 lines_

```
[...truncated 8 lines...]
 > git -c core.askpass=true fetch --tags --progress https://github.com/TheInnerCircleO/hangupsbot +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/16/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/16/merge^{commit} # timeout=10
Checking out Revision be7e5d86c8770baa25e68435e9432470e131cc40 (refs/remotes/origin/pr/16/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f be7e5d86c8770baa25e68435e9432470e131cc40
First time build. Skipping changelog.
[workspace] $ /bin/bash /tmp/hudson4885241598160297720.sh
pep8 recreate: /var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8
pep8 installdeps: -r/var/lib/jenkins/jobs/hangups_tox/workspace/requirements.txt, -r/var/lib/jenkins/jobs/hangups_tox/workspace/test-requirements.txt
pep8 develop-inst: /var/lib/jenkins/jobs/hangups_tox/workspace
pep8 runtests: PYTHONHASHSEED='1237644907'
pep8 runtests: commands[0] | flake8
./hangupsbot/commands.py:489:24: W291 trailing whitespace
                i['t'],
                       ^
./hangupsbot/commands.py:495:80: E501 line too long (91 > 79 characters)
            result.append(hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK))
                                                                               ^
ERROR: InvocationError: '/var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8/bin/flake8'
___________________________________ summary ____________________________________
ERROR:   pep8: commands failed
Build step 'Execute shell' marked build as failure
Setting status of 1a96086918f1f108f38dd0f3ddca7edd338cb4bb to FAILURE with url http://bock.systems:8080/job/hangups_tox/21/ and message: Merged build finished.
```

Oh No...

**Comment by Penguin2600** [comment]
maxresdefault

**Comment by uberscientist** [comment]
I am so mad

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/22/

**Build Log**
_last 25 lines_

```
[...truncated 5 lines...]
 > git config remote.origin.url https://github.com/TheInnerCircleO/hangupsbot # timeout=10
Fetching upstream changes from https://github.com/TheInnerCircleO/hangupsbot
 > git --version # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/TheInnerCircleO/hangupsbot +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/16/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/16/merge^{commit} # timeout=10
Checking out Revision d68c288c7980ad8fa8dcd54b5e2628d815fed1be (refs/remotes/origin/pr/16/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f d68c288c7980ad8fa8dcd54b5e2628d815fed1be
First time build. Skipping changelog.
[workspace] $ /bin/bash /tmp/hudson7266054848896174126.sh
pep8 recreate: /var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8
pep8 installdeps: -r/var/lib/jenkins/jobs/hangups_tox/workspace/requirements.txt, -r/var/lib/jenkins/jobs/hangups_tox/workspace/test-requirements.txt
pep8 develop-inst: /var/lib/jenkins/jobs/hangups_tox/workspace
pep8 runtests: PYTHONHASHSEED='2631260499'
pep8 runtests: commands[0] | flake8
./hangupsbot/commands.py:495:80: E501 line too long (91 > 79 characters)
            result.append(hangups.ChatMessageSegment('\n', hangups.SegmentType.LINE_BREAK))
                                                                               ^
ERROR: InvocationError: '/var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8/bin/flake8'
___________________________________ summary ____________________________________
ERROR:   pep8: commands failed
Build step 'Execute shell' marked build as failure
Setting status of 626e88d9cd39ce9af839f40ee43706553e4a7a42 to FAILURE with url http://bock.systems:8080/job/hangups_tox/22/ and message: Merged build finished.
```

Oh No...

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/23/
Tests Passed, Yay.

**Comment by uberscientist** [comment]
ok now gimme ur :+1:s where they count

File: hangupsbot/commands.py [hunk 1]
```diff
@@ -483,9 +484,17 @@ def stock(bot, event, *args):
...
         data = json.loads(raw_data.text[3:])
-        result = '\n'.join([i['t'] + ": " + i['l'] for i in data])
+        for i in data:
+            ticker_link = hangups.ChatMessageSegment(
+                i['t'],
+                hangups.SegmentType.LINK,
+                link_target="https://www.google.com/finance?q=" + i['t']
+            )
+            result.append(ticker_link)
+            result.append(': ' + i['l'] + ' (' + i['cp'] + '%)')
```
Comment (PHLAK) [inline hangupsbot/commands.py]: Shouldn't this be:

```
result.append(
    hangups.ChatMessageSegment(': ' + i['l'] + ' (' + i['cp'] + '%)')
)
```

Could even improve it more with `.format()`:

```
result.append(
    hangups.ChatMessageSegment(': {} ({}%)'.format(i['l'], i['cp']))
)
```

**Comment by Penguin2600** [comment]
:imp:  :godmode: :+1: :shipit:

File: hangupsbot/commands.py (hunk 1 above)
Comment (DavidCaudill) [inline hangupsbot/commands.py]: Maybe, unless it defaults to a text segment. I hope this was tested?

**Comment by PHLAK** [comment]
:penguin: (aside from my comment above)

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/24/

**Build Log**
_last 25 lines_

```
[...truncated 8 lines...]
 > git -c core.askpass=true fetch --tags --progress https://github.com/TheInnerCircleO/hangupsbot +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/16/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/16/merge^{commit} # timeout=10
Checking out Revision 309292b5203088cf04a1e3e8018223e0e2e6707d (refs/remotes/origin/pr/16/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 309292b5203088cf04a1e3e8018223e0e2e6707d
First time build. Skipping changelog.
[workspace] $ /bin/bash /tmp/hudson5075942264958506014.sh
pep8 recreate: /var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8
pep8 installdeps: -r/var/lib/jenkins/jobs/hangups_tox/workspace/requirements.txt, -r/var/lib/jenkins/jobs/hangups_tox/workspace/test-requirements.txt
pep8 develop-inst: /var/lib/jenkins/jobs/hangups_tox/workspace
pep8 runtests: PYTHONHASHSEED='2182619249'
pep8 runtests: commands[0] | flake8
./hangupsbot/commands.py:494:13: E265 block comment should start with '# '
            #result.append(': ' + i['l'] + ' (' + i['cp'] + '%)')
            ^
./hangupsbot/commands.py:495:13: F821 undefined name 'results'
            results.append(
            ^
ERROR: InvocationError: '/var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8/bin/flake8'
___________________________________ summary ____________________________________
ERROR:   pep8: commands failed
Build step 'Execute shell' marked build as failure
Setting status of e7e093a33067ddbba7f567180c892a493f542d82 to FAILURE with url http://bock.systems:8080/job/hangups_tox/24/ and message: Merged build finished.
```

Oh No...

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/25/
Tests Passed, Yay.

**Comment by Penguin2600** [comment]
:gemini: :star2: :v:

**Comment by System-Flenser** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/26/
Tests Passed, Yay.

**Comment by uberscientist** [comment]
:capricorn: :cool: :sunglasses: :+1:

**Pull request merged**